lib/engine/game/g_18_christmas_eve/step/buy_sell_par_gift_shares.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/buy_sell_par_shares'
+
+module Engine
+  module Game
+    module G18ChristmasEve
+      module Step
+        class BuySellParGiftShares < Engine::Step::BuySellParShares
+          def description
+            'Sell/Buy/Sell Shares and/or Gift Certs'
+          end
+
+          def round_state
+            super.merge!(
+              {
+                presidencies_gifted: [],
+              }
+            )
+          end
+
+          def actions(entity)
+            return [] unless entity == current_entity
+
+            actions = super
+            actions << 'choose' if can_gift_any?
+            actions << 'pass' unless actions.empty? && actions.include?('pass')
+            actions
+          end
+
+          def choice_available?(entity)
+            return false unless @game.sellable_turn? && entity.corporation?
+
+            @choices = gift_choices(entity)
```

Comment:
you shouldn't be assigning @choices within a question mark method

Reply:
I think I agree. I'm trying to come up with a better way - but in `stock.rb`:160, `choices` is called for corporations in line 11 of `choose.rb` without any parameters. `choice_available?` is the only function called with each corporation as a parameter - meaning that unless I change `choose.rb` and `stock.rb` to include an entity parameter, the choices have to be the same for each corporation (albeit - they can be shown on a per-corporation basis). 

In this case - I was trying to limit the impact of my changes outside of 18ChEv's classes as much as possible. In principal - would me making the change to `stock.rb` and `choose.rb` be acceptable? Or, should I add a 'company specific choice' to `stock.rb`, or was there something else - like another action - I could have used other than choices that would have been more appropriate?

Reply:
i don't understand why you need to set @choices, i don't have much context

Reply:
A president of a corp may gift a cert to another player. They may gift only to a player that has none of that corp. They may gift a presidents certificate, if that player will have the most or equal most shares after gifting. The result of this is that the choices for one corp may be ['Gift cert to Player 1', 'Gift cert to Player 2'] and another corp may be ['Give prez to Player 3']. Choices shows what buttons are available for the `choose` action. If I don't change `@choices` each time (or change it into a function that accepts an entity), then the same buttons will appear for each company.

Reply:
For instance:
![image](https://user-images.githubusercontent.com/478240/148670787-c3f055e6-8632-47df-aa1a-9e9367222716.png)

Reply:
shouldn't the buttons only appear when you click on it though? you shouldn't have buttons appearing on corps that aren't selected

Reply:
You are right - I thought they should too, but didn't seem to be a supported way of doing it with `choose`/`choice` - and so I thought, at least while alpha, this might be acceptable. The 1828 use of `choose` for merge uses merge specific code hiding in the view. Was trying to limit the scope of my changes (in part because I was nervous that putting company `choice` options in the corporation hidey button space would effect another game) - but I'll spend some time and see whether it will effect any other games.

Reply:
I've attempted to resolve both of these in https://github.com/tobymao/18xx/pull/6732/commits/a344f4bd905881f5c04f6399f40e5329646489fd 
I think it's an appropriate approach - but happy to hear if you think (for instance) that there should be a whole new class of `choose` action, and want me to back this back out. 
I was honestly surprised that adding specific actions to the stock round for specific companies was novel in the code - I'd still be glad to hear that I missed something, and there was something else I could have used.